User can view their entries

t-doan opened the pull request:
![user-can-view-their-entries](https://github.com/t-doan/code-journal/assets/26643555/5fac2514-a25b-4a72-8244-f1a7930c79f2)

t-doan committed:
Add an unordered list to the div entries element
Style the entries in the list to match with the figma
The layout is also responsive to mobile and desktop screens #2

t-doan committed:
Create a function that will take an argument entry and it will generate and return a dom tree fo the single entry. #2

t-doan committed:
Add an event listener to the document to which it will start a function that will iterate the array of data entries, nreder them, and append them to the ul #2

t-doan committed:
Added a toggle to which show/hide the noEntry text #2

t-doan committed:
fix eslint

t-doan committed:
Have a function that switch view between the entries and the form #2

t-doan committed:
Add two event listener to change views on NEW button and Entries #2

t-doan committed:
Able to switch views.
When submitting, toggle to the entries view and see recent entries and remove the noEntry message
Test in private window.
When refreshing the page, is able to stay on the entries page rather than back to the new entry view. #2

shawnkost commented on css/styles.css:
```diff
@@ -1,3 +1,4 @@
+/* stylelint-disable string-quotes */
```
Delete this. Don't add lint rules without asking. We have rules setup for a reason

shawnkost commented on js/main.js:
```diff
@@ -25,4 +35,71 @@ function handleSubmit(event) {
+  }
+}
+
+function toggleNoEntries() {
+  const $h1 = document.createElement('h1');
+  if (data.entries.length === 0) {
+    $h1.innerText = 'THERE ARE NO ENTRIES RECORDED';
+    $ul.appendChild($h1);
+  } else {
+    const existing = $ul.querySelector('h1');
+    if (existing) {
+      existing.remove();
+    }
+  }
+}
```
This text should be in your HTML file. The only purpose of this function is to check the length of `data.entries` and either show or hide the element accordingly. Not create it and delete it

shawnkost commented on js/main.js:
```diff
@@ -25,4 +35,71 @@ function handleSubmit(event) {
+
+function viewSwap(view) {
+  if (view === 'entries') {
+    document.getElementById('entries').classList.remove('hidden');
+    document.getElementById('entryForm').className = 'hidden';
+    data.view = view;
+  } else if (view === 'entryForm') {
+    document.getElementById('entryForm').classList.remove('hidden');
+    document.getElementById('entries').className = 'hidden';
```
You want to try and do your `querySelector`s globally for the most part. By doing them inside of a function, you're having to traverse the DOM tree over and over every time this function gets called. By moving this outside of the function, then you do the search once and always have a reference to it

shawnkost reviewed (changes requested):
Few changes here Tai!

t-doan committed:
made changes.
Removed stylelint
Modify toggleNoEntries to check only if data.entries is 0, and moved text to html file.
Chnage viewSwap() to use global variables #2

t-doan commented on css/styles.css:
```diff
@@ -1,4 +1,3 @@
-/* stylelint-disable string-quotes */
```
made some changes.

t-doan commented on index.html:
```diff
@@ -69,6 +69,7 @@ <h1>New Entry</h1>
           <h1>Entries</h1>
           <a href="#" class="entryForm"><h3 id="new-button">NEW</h3></a>
         </div>
+        <h1 id="noEntry" class="hidden">THERE ARE NO ENTRIES RECORDED</h1>
```
add the no entries to html file

t-doan commented on js/main.js:
```diff
@@ -6,6 +6,9 @@ const $formSubmit = document.querySelector('#entry-form');
 const $ul = document.querySelector('ul');
 const $entry = document.querySelector('.entry');
 const $entryForm = document.querySelector('.entryForm');
+const $noEntry = document.querySelector('#noEntry');
+const $entriesView = document.querySelector('#entries');
+const $entryFormView = document.querySelector('#entryForm');
```
Created global variables to the functions

t-doan commented on js/main.js:
```diff
@@ -80,26 +83,22 @@ function handleContent(event) {
-    if (existing) {
-      existing.remove();
-    }
+    $noEntry.setAttribute('class', 'hidden');
```
Change to using a global variable and to only check for the data.entries.length and show/hide the message

t-doan commented on js/main.js:
```diff
@@ -80,26 +83,22 @@ function handleContent(event) {
-    document.getElementById('entryForm').classList.remove('hidden');
-    document.getElementById('entries').className = 'hidden';
+    $entryFormView.classList.remove('hidden');
+    $entriesView.className = 'hidden';
```
Changed it all to use a global variable

t-doan committed:
Removed old console.log and fixed styles.css stlying due to eslint #2

shawnkost reviewed (approved):
👍🏻
